manager-bundle/src/Twig/FileExtensionFilterIterator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerBundle\Twig;
+
+class FileExtensionFilterIterator implements \IteratorAggregate
+{
+    /**
+     * @var \Traversable
+     */
+    private $iterator;
+
+    public function __construct(\IteratorAggregate $templateIterator)
+    {
+        $this->iterator = $templateIterator->getIterator();
+    }
+
+    public function getIterator(): \CallbackFilterIterator
+    {
+        return new \CallbackFilterIterator(
+            new \IteratorIterator($this->iterator),
+            function ($path): bool {
+                return $this->acceptsPath($path);
+            }
+        );
+    }
+
+    /**
+     * Filter files by extension (in the root namespace).
+     */
+    private function acceptsPath(string $path): bool
+    {
+        return 0 === strncmp($path, '@', 1) || '.twig' === substr($path, -5);
```

Comment:
Because then it's clear what this functional expression does: whether a path is accepted or not. This style is imo favorable over comments. :man_shrugging:

Reply:
But it's up to you :smile:  If inlining is more consistent with the current code style, let's do that oc.